logging/src/jvmMain/kotlin/io/spine/logging/LoggingFactory.kt

```diff
@@ -88,12 +88,14 @@ public actual object LoggingFactory: ClassValue<JvmLogger>() {
     }
 
     private fun createForClass(cls: Class<*>): JvmLogger {
-        val impl = createFluentLogger(cls)
-        return JvmLogger(cls.kotlin, impl)
+        val floggerBackend = Platform.getBackend(cls.name)
+        val flogger = FluentLogger2(floggerBackend)
+        // As for now, `JvmLogger` just delegates actual work to Flogger.
+        return JvmLogger(cls.kotlin, flogger)
     }
 
-    private fun createFluentLogger(cls: Class<*>): FluentLogger2 {
-        val backend = Platform.getBackend(cls.name)
-        return FluentLogger2(backend)
+    @JvmStatic
+    public actual fun <API : LoggingApi<API>> forEnclosingClass(): Logger<API> {
+        return loggerFor(this::class)
```

Comment:
@yevhenii-nadtochii, how come this works in a static context _and_ in non-static context?

When we do `loggerFor(this::class)` in, say `WithLogging`, we get the class which implements this interface. Here we get `LoggingFactory::class`.

Reply:
Good catch! Turns out, it doesn't work as expected. The same logger is always returned. I will think how to better test it in #38, beside checking different classes get different loggers.